### PR TITLE
Enable build of ARM64 installers in ASP.NET Core

### PR DESCRIPTION
**Pull request by hoyosjs**
Enable ARM64 installers build.

- Changes WiX toolset used to 3.14 to support ARM64
- Generates targeting pack from the x86/x64 leg, as it gets produced using a zip that gets generated there.
- The ARM leg now produces all the necessary msi's, exe, and wixlib needed for the installer to generate a bundle.

### CODE REVIEW

File: NuGet.config
Comment (hoyosjs) [inline NuGet.config:14]: I added this feed for testing. The main issue was that WiX doesn't have a package for the toolset that supports ARM64. Other repositories don't use `wixproj`, so we didn't need it before this. I don't know what feed this should go to, or if this is even a good idea.

File: eng/Build.props
Comment (hoyosjs) [inline eng/Build.props:69]: This was the other sort of... not-so-clean thing that was necessary. The right solution here is to stop using . This was more straightforward and felt less risky (given this change needs to go to 5.0).

**Comment by hoyosjs** [comment]
Official test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=800872&view=results

Further validation: 
- Confirmed files are R2R, signed files.
- Now checking registry keys and signatures of the bundles. 

cc: @joeloff, @tommcdon

**Comment by Pilchie** [comment]
Do we want to target this to `release/5.0-rc2`?

**Comment by hoyosjs** [comment]
@pilchie, yes. I will port once this gets reviewed and goes in.

**Comment by joeloff** [comment]
@Pilchie if we did, we might be able to unblock the SDK and get that part completed for RC2 as well, but it would be tight. I'll have a chat with other SDK folks. @marcpopMSFT as an FYI

**Comment by tommcdon** [comment]
FYI @richlander

File: src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj
Comment (dougbu) [inline src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj:58]: Are we producing a completely separate shared Fx bundle for ARM64 because the platform-specific framework is created in a different job❔ Could the extra bundle confuse customers❔

**Review by dougbu — CHANGES_REQUESTED** [review]

File: NuGet.config
Comment (hoyosjs) [reply NuGet.config:14]: Yeah, this was to verify this won't break in CI. I'm asking for some guidance here.

File: src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj
Comment (hoyosjs) [reply src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj:58]: After discussion, we decided it doesn't make sense to ship an exe bundle with only the ARM64 installer. The x86+arm64 bundle doesn't provide enough value, and we've deferred this for 6.0 as this is a PR to port for RC2

**Comment by dougbu** [comment]
I can't approve this until the `general-testing` feed is removed. @hoyosjs you've addressed my other comments well. (nit: "only" isn't spelled "inly" even in comments.)

Also suggest @joeloff or someone else w/ more Wix knowledge have a look at these changes. They follow the existing patterns but I may be missing something.

File: NuGet.config
Comment (hoyosjs) [reply NuGet.config]: @dougbu I got the green light to move this as is to `dotnet-eng`. I've reverted all changes to `NuGet.config`

**Review by dougbu — APPROVED** [review]
I'm fine with this but would really appreciate additional eyes on it @joeloff @dotnet/aspnet-build or anyone w/ more WiX experience than me, please chime in.

**Comment by joeloff** [comment]
Looks good. I'm guessing that the hosting bundle will remain x86/x64 only for the time being. Unlike the other installers, the hosting bundle prefers to install all architectures by default and I think there will be a clash since for x64, there aren't new properties defined yet (at least none that I can see in Windows).

My big concern is the wixlib because we rely on that to share the package chaining across the shared framework, hosting and SDK installer bundles.

**Comment by hoyosjs** [comment]
@joeloff any verification I can do to verify? I have official runs so I have signed installers and can test anything that's needed. However, this is becoming time sensitive.

**Review by joeloff — APPROVED** [review]

**Comment by joeloff** [comment]
@hoyosjs if the have the signed MSI, you can try them out on a VM. You can also double check the Product/Upgrade code for the x64/arm64 MSIs just to be sure they are different (you can use ORCA for that). The build seeds the GUIDs using the Platform property, so they should be different.

**Comment by hoyosjs** [comment]
@joeloff I've tested this:

image

> My big concern is the wixlib because we rely on that to share the package chaining across the shared framework, hosting and SDK installer bundles.

Do you have a particular concern I can check against?

**Comment by hoyosjs** [comment]
Issues I keep hitting are https://github.com/dotnet/aspnetcore/issues/25623 and a gradle issue solved by https://github.com/dotnet/aspnetcore/pull/25707. Closing and reopening to create a rebase.

**Comment by hoyosjs** [comment]
@dougbu I've tried rebasing but I keep getting red. Looking around I see you had a comment describing all the issues I've seen. However they seem to have gotten fixes upstream, but they keep failing. Any advice?

> * #25651 hit another flaky Razor SDK test (`BundlesScopedCssFiles_UpdatesBundleWhenContentsChange`)   
>   * this test failure impacted most, if not all, of these PRs
> * #25652 hit the PDB duplication build problem in both aspnetcore-ci and aspnetcore-quarantined-pr
> * #25653 hit the `gradlew` build problem @halter73
> * #25654 hit the PDB duplication build problem but only in the aspnetcore-quarantined-pr pipeline

https://github.com/dotnet/aspnetcore/issues/25293#issuecomment-687691562

**Comment by dougbu** [comment]
@hoyosjs the problem here is you're targeting the master branch and all the fixes mentioned have been checked into release/5.0-rc2. Need to either wait for #25650 to merge or target release/5.0-rc2 instead. Recommend doing the second because time for RC2 is getting short.

**Comment by hoyosjs** [comment]
@dougbu I didn't know there was an automated FI from RC2 to master. I've retargeted this to RC2.

**Comment by hoyosjs** [comment]
Feel free to merge once it's green

@Pilchie can this please be considered for RC2 for the ARM64 installer?

**Comment by dougbu** [comment]
All checks 🟢 Waiting on your word @Pilchie&hellip;

**Comment by Pilchie** [comment]
Just wanted to double check with Tactics - we're good to merge this now.

**Comment by Pilchie** [comment]
Approved for .NET 5 RC2.

**Comment by dougbu** [comment]
Thanks very much @hoyosjs❕

**Pull request merged**
